cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -501,6 +512,13 @@ private boolean checkProgramsRunning(final NamespaceId namespaceId) {
     return !store.getActiveRuns(namespaceId).isEmpty();
   }
 
+  private boolean checkUsedInTethering(NamespaceId namespaceId) throws IOException {
```

Comment:
instead of a boolean, we should return the set of problematic connections. That way the error message can be more useful and include the problematic connections

Reply:
fixed